File: DotNet/CesiumLanguageWriter/Advanced/ICesiumCartesian3VelocityValuePropertyWriter.cs
using System.Collections.Generic;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a three-dimensional Cartesian value and its derivative.
    /// </summary>
    public interface ICesiumCartesian3VelocityValuePropertyWriter : ICesiumInterpolatablePropertyWriter
    {
        /// <summary>
        /// Writes the value as a three-dimensional Cartesian value and its derivative.
        /// </summary>
        /// <param name="value">The value.</param>
        void WriteCartesianVelocity(Motion<Cartesian> value);

        /// <summary>
        /// Writes the value as a three-dimensional Cartesian value and its derivative.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        void WriteCartesianVelocity(IList<JulianDate> dates, IList<Motion<Cartesian>> values);

        /// <summary>
        /// Writes the value as a three-dimensional Cartesian value and its derivative.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        /// <param name="startIndex">The index of the first element to write.</param>
        /// <param name="length">The number of elements to write.</param>
        void WriteCartesianVelocity(IList<JulianDate> dates, IList<Motion<Cartesian>> values, int startIndex, int length);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumCartesian3VelocityValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import java.util.List;
import javax.annotation.Nonnull;

/**
 * A writer that can write a value as a three-dimensional Cartesian value and its derivative.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public interface ICesiumCartesian3VelocityValuePropertyWriter extends ICesiumInterpolatablePropertyWriter {
    /**
    * Writes the value as a three-dimensional Cartesian value and its derivative.
    * @param value The value.
    */
    void writeCartesianVelocity(@Nonnull Motion1<Cartesian> value);

    /**
    * Writes the value as a three-dimensional Cartesian value and its derivative.
    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    */
    void writeCartesianVelocity(List<JulianDate> dates, List<Motion1<Cartesian>> values);

    /**
    * Writes the value as a three-dimensional Cartesian value and its derivative.
    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    * @param startIndex The index of the first element to write.
    * @param length The number of elements to write.
    */
    void writeCartesianVelocity(List<JulianDate> dates, List<Motion1<Cartesian>> values, int startIndex, int length);
}